crates/wasmtime/src/module.rs

```diff
@@ -307,6 +266,19 @@ impl Module {
     /// # }
     /// ```
     pub fn from_binary(engine: &Engine, binary: &[u8]) -> Result<Module> {
+        if let Some(module) = SerializedModule::from_bytes(binary)? {
+            return module.into_module(engine);
+        }
+
+        // Check to see that the config's target matches the host
+        let target = engine.config().isa_flags.triple();
+        if *target != target_lexicon::Triple::host() {
+            bail!(
+                "target '{}' specified in the configuration does not match the host",
+                target
+            );
+        }
```

Comment:
Technically I think we might want to check a number of things here in the limit, e.g. if you enable avx512 or something like that but you don't have it we should probably bail out here at some point. I'm fine with that being a FIXME though to basically check things other than just the target.